app/controllers/zones_controller.rb

```diff
@@ -7,9 +7,12 @@ class ZonesController < ApplicationController
   load_and_authorize_resource
 
   def show
+    @planning = params.key?(:planning_id) ? current_user.customer.plannings.find(params[:planning_id]) : nil
     if params.key?(:destination_ids)
       destination_ids = params[:destination_ids].split(',')
       @destinations = current_user.customer.destinations.where(ParseIdsRefs.where(Destination, destination_ids))
+    elsif @planning
```

Comment:
Compute only @planning if we are in the else case to avoid compute and query database it if not required ?